pandas/core/groupby/generic.py

```diff
@@ -1750,13 +1753,32 @@ def value_counts(
             if sort:
                 # Sort the values and then resort by the main grouping
                 index_level = range(len(self.grouper.groupings))
-                result = result.sort_values(ascending=ascending).sort_index(
-                    level=index_level, sort_remaining=False
+                result = (
+                    cast(Series, result)
+                    .sort_values(ascending=ascending)
+                    .sort_index(level=index_level, sort_remaining=False)
                 )
 
             if not self.as_index:
                 # Convert to frame
-                result = result.reset_index(name="proportion" if normalize else "count")
+                name = "proportion" if normalize else "count"
+                columns = result.index.names
+                if name in columns:
+                    raise ValueError(
+                        f"Column label '{name}' is duplicate of result column"
+                    )
+                columns = com.fill_missing_names(columns)
+                values = result.values
```

Comment:
result is a Series at this point?  ._values is generally preferable to .values, as the latter will cast dt64tz to ndarray[object]

Reply:
Thanks!
I decided to do this differently following a suggestion from a while ago (https://github.com/pandas-dev/pandas/pull/44755#issuecomment-986258740)
I really wanted these methods to be public, but it is not happening...
Anyway, this way is more DRY and I shouldn't really be writing my own version of reset_index().